algviz/interface/visitors.py

```diff
@@ -190,3 +190,29 @@ class WidgetVisitor(Visitor):
 
     def visit(self, *args, **kwargs):
         return super().visit(*args, **kwargs)
+
+class LinkedListVisitor(Visitor):
+    """Linked lists are represented as an array of an array and a current index, i.e.,
+    [[1, 2, 3], 0]."""
+    type_ = Tokens.ARRAY_T#treat this as an array, at least for now.
```

Comment:
Linked Lists shouldn't be arrays.  They should be BinaryTrees if they're anything.  Looks like either you or I should add a LinkedList to structures.py and json_objects.py.

Reply:
I can work on this in the coming week